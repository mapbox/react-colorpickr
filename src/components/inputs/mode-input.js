'use strict';

import React from 'react';
import PropTypes from 'prop-types';

class ModeInput extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    checked: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired
  };

  render() {
    const { name, checked, onChange } = this.props;
    return (
      <div className='flex-child flex-child--no-shrink w24 flex-parent flex-parent--center-cross'>
        <input className='cursor-pointer' type="radio" name={name} checked={checked} onChange={onChange} />
      </div>
    );
  }
}

export default ModeInput;
