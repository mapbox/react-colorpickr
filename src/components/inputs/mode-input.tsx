import React from 'react';
import PropTypes from 'prop-types';
import themeable from 'react-themeable';
import { autokey } from '../../autokey.ts';

function ModeInput({ theme, checked, onChange, readOnly }) {
  const themer = autokey(themeable(theme));
  return (
    <div {...themer('modeInputContainer')}>
      <input
        {...themer('modeInput')}
        type="radio"
        checked={checked}
        onChange={onChange}
        disabled={readOnly}
      />
    </div>
  );
}

ModeInput.defaultProps = {
  readOnly: false
};

ModeInput.propTypes = {
  name: PropTypes.string.isRequired,
  theme: PropTypes.object.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  readOnly: PropTypes.bool
};

export { ModeInput };
