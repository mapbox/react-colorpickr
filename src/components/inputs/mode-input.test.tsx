import React from 'react';
import { render, screen } from '@testing-library/react';
import { ModeInput } from './mode-input';
import userEvent from '@testing-library/user-event';

describe('ModeInput', () => {
  let wrapper;
  const props = {
    id: 'mode',
    checked: false,
    theme: {},
    name: 'mode',
    onChange: jest.fn()
  };

  beforeEach(() => {
    wrapper = render(<ModeInput {...props} />);
  });

  test('renders', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  test('onChange', () => {
    const input = screen.getByRole('radio');
    userEvent.click(input);
    expect(props.onChange).toHaveBeenCalled();
  });
});
