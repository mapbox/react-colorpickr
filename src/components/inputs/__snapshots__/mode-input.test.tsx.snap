// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeInput renders 1`] = `
<body>
  <div>
    <div>
      <input
        type="radio"
      />
    </div>
  </div>
</body>
`;
